src/graph.rs

```diff
@@ -1138,14 +1105,21 @@ impl<'a> Iterator for ModuleGraphErrorIterator<'a> {
 /// which can be located in the module "slots" in the graph. The graph also
 /// contains any redirects where the requested module specifier was redirected
 /// to another module specifier when being loaded.
-#[derive(Debug, Default, Clone)]
+#[derive(Debug, Default, Clone, Serialize)]
 pub struct ModuleGraph {
+  #[serde(skip_serializing)]
   graph_kind: GraphKind,
   pub roots: Vec<ModuleSpecifier>,
+  #[serde(rename = "modules")]
+  #[serde(serialize_with = "serialize_module_slots")]
   pub(crate) module_slots: BTreeMap<ModuleSpecifier, ModuleSlot>,
+  #[serde(skip_serializing_if = "BTreeMap::is_empty")]
+  #[serde(serialize_with = "serialize_graph_imports")]
```

Comment:
Do these need to use two separate attributes? Can't they be combined onto a single attribute call?

Reply:
You can use comma syntax as well but it would just be four lines instead of two and imo better to separate them like imports. Or if you mean putting the skipping logic in the function, I don't think that's possible because it can't control key inclusion